runtime/doc/quickfix.txt

```diff
@@ -1921,6 +1922,63 @@ error messages into a format that quickfix mode will understand.  See the
 start of the file about how to use it.  (This script is deprecated, see
 |compiler-perl|.)
 
-
+=============================================================================
+10. Customizing the quickfix window		*quickfix-window-function*
+
+The default format for the lines displayed in the quickfix window and location
+list window is:
+
+    <filename>|<lnum> col <n>|<text>
+
+This can be customized by setting the 'quickfixtextfunc' option to a Vim
```

Comment:
worth introducing the "why" early - the what is fine, but not the why - e.g. what's a motivating use case for this ?

Reply:
There have been asks in the past for customizing the quickfix buffer contents.

For example, you search for a pattern from the top of a source tree ,
and populate the quickfix list with the search results (using a tool like ripgrep).
If you change the current directory now (because you have set 'autochdir' or
used the lcd or tcd commands), then the quickfix window will show the 
complete path to the files. If the source tree is deeply nested, then the
results in the quickfix window will be hard to read. It will be useful in
this scenario to abbreviate the file path and show only the last few
characters to to show SRCROOT to replace the path before the
project root directory.

Similarly there was an ask to show the module information in the
quickfix buffer. This was supported by modifying the Vim source
code. Instead of modifying the source code, we can use this
function to customize the quickfix fix line.

Reply:
An example to illustrate how to substitute the project source root directory
with SRCROOT in the quickfix buffer is below:

func Myqffunc(info)
    " get the quickfix list entry
    let e = getqflist({'id' : a:info.id, 'idx' : a:info.idx,
		\ 'items' : 1}).items[0]
    " substitute the source root directory with SRCROOT
    let s = fnamemodify(bufname(e.bufnr),
		\ ':p:.:s#/project/source/root#SRCROOT#')
    " add the line number and the line text
    let s ..= '|' .. e.lnum .. '| ' .. substitute(e.text, '^\s\+', '', '')
    return s
endfunc
set quickfixtextfunc=Myqffunc